config/formats.ts

```diff
@@ -4478,6 +4478,114 @@ export const Formats: import('../sim/dex-formats').FormatList = [
+			'Nickname Clause', 'Endless Battle Clause', 'Cancel Mod', 'Picked Team Size = 4', 'Exact HP Mod', "Item Clause = 1", 'Open Team Sheets'],
+		banlist: ['Soul Dew', 'Deoxys-Defense', 'Deoxys-Attack', 'Deoxys-Speed'],
+		unbanlist: ['Latios', 'Latias', 'Wobbuffet', 'Wynaut'],
+		restricted: ['Restricted Legendary', "Mythical"],
+		bestOfDefault: true,
+		rated: true,
+		battle: {
+			tiebreak() {
+				if (this.ended) return false;
+
+				this.inputLog.push(`>tiebreak`);
+				this.add('message', "Time's up! Going to tiebreaker...");
+				const notFainted = this.sides.map(side => (
+					side.pokemon.filter(pokemon => !pokemon.fainted).length
+				));
+				this.add('-message', this.sides.map((side, i) => (
+					`${side.name}: ${notFainted[i]} Pokemon left`
+				)).join('; '));
+				const maxNotFainted = Math.max(...notFainted);
+				let tiedSides = this.sides.filter((side, i) => notFainted[i] === maxNotFainted);
+				if (tiedSides.length <= 1) {
+					return this.win(tiedSides[0]);
+				}
+
+				const hpPercentage = tiedSides.map(side => (
+					side.pokemon.map(pokemon => pokemon.hp / pokemon.maxhp).reduce((a, b) => a + b) * 100 / 6
+				));
+				this.add('-message', tiedSides.map((side, i) => (
+					`${side.name}: ${Math.round(hpPercentage[i])}% total HP left`
+				)).join('; '));
+				const maxPercentage = Math.max(...hpPercentage);
+				tiedSides = tiedSides.filter((side, i) => hpPercentage[i] === maxPercentage);
+				if (tiedSides.length <= 1) {
+					return this.win(tiedSides[0]);
+				}
+
+				return this.tie();
+			},
+		},
+		onBeforeTurn(pokemon) {
+			if (this.turn === 15 && this.effectState.printed === undefined && this.field.getPseudoWeather("timer")) {
+				this.effectState.printed = true;
+				this.add(`raw|<div class="broadcast-red"><b>The Battle ends after turn 18!</b></div>`);
+			}
+			if (this.turn > 18 && this.field.getPseudoWeather("timer")) {
+				let allyAlive = 0;
+				let allyPer = 0;
+
+				let foeAlive = 0;
+				let foePer = 0;
+
+				for (const poke of pokemon.side.pokemon) {
+					if (!poke.fainted) {
+						allyAlive += 1;
+						allyPer += (poke.hp / poke.maxhp);
+					}
+				}
+				for (const poke of pokemon.side.foe.pokemon) {
+					if (!poke.fainted) {
+						foeAlive += 1;
+						foePer += (poke.hp / poke.maxhp);
+					}
+				}
+
+				this.hint("18 turn limit reached. Deciding the winner");
+				if (allyAlive > foeAlive) {
+					this.hint(`${pokemon.side.name} had more pokemon`);
+					this.forceWin(pokemon.side.id);
+				} else if (allyAlive < foeAlive) {
+					this.hint(`${pokemon.side.foe.name} had more pokemon`);
+					this.forceWin(pokemon.side.foe.id);
+				} else if (allyPer > foePer) {
+					this.hint(`${pokemon.side.name} had a higher total hp percentage`);
+					this.forceWin(pokemon.side.id);
+				} else if (allyPer < foePer) {
+					this.hint(`${pokemon.side.foe.name} had a higher total hp percentage`);
```

Comment:
Can you implement HP decider as a method in ``sim.ts``?

Reply:
Done